p2p/host/basic/basic_host.go

```diff
@@ -414,7 +414,7 @@ func (h *BasicHost) newStreamHandler(s network.Stream) {
 			}
 			logf("protocol EOF: %s (took %s)", s.Conn().RemotePeer(), took)
 		} else {
-			log.Debugf("protocol mux failed: %s (took %s)", err, took)
+			log.Debugf("protocol mux failed: %s (took %s, id:%s)", err, took, s.ID())
```

Comment:
We can add details from the connection here. 
`s.Conn().RemotePeer()` and `s.Conn().RemoteMultiAddr()`

Reply:
Done.